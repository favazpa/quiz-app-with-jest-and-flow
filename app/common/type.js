//@flow

import type {
  TextStyleProp,
  ViewStyleProp,
} from 'react-native/Libraries/StyleSheet/StyleSheet';

type ScoreType = {|
  score?: number ,
  handleRetry?: function,
|};

type QuizNumberType = {|
  style?: TextStyleProp,
  quizes?: Array<any>,
  currentQuestionNumber?: number,
  totalQuizNumber?: number,
|};

type QuestionType = {|
  data?: Object,
|};

type OptionTypes = {|
  questionNumber?:number,
  option?:string,
  selected?: boolean,
  onPress?: function,
  key?:number,
|};

type DividerTypes = {|
  style?: ViewStyleProp,
|};

type ButtonTypes = {|
  title?: string,
  tilteStyle?: TextStyleProp,
  onPress?: () => Promise<void> | void,
  style?: ViewStyleProp,
|};

type OptionsTypes = {|
  options?:Array<any>,
  checkSelection?:function,
  onPressOption?:function
  |}

export type {
  ScoreType,
  QuizNumberType,
  QuestionType,
  OptionTypes,
  DividerTypes,
  ButtonTypes,
  OptionsTypes
};
