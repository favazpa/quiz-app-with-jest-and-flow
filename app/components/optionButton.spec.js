// @flow
import React from 'react';
import {Component} from 'react';
import OptionButton from './optionButton';
import ShallowRenderer from 'react-test-renderer/shallow';
import {findWithType, findAllWithType} from 'react-shallow-testutils';
import type {OptionTypes} from '../common/type';
import {TouchableOpacity} from 'react-native';

let renderer = new ShallowRenderer();

const renderButton = () => {
  renderer.render(<OptionButton selected={true} />);
  return renderer.getRenderOutput();
};
describe('src/component/options', () => {
  test('should render correctly', () => {
    const result = renderButton();
    expect(result).toMatchSnapshot();
  });

  test('should press option button', () => {
    const result = renderButton();
    const btn = findWithType(result, TouchableOpacity);
    btn.props.onPress;
  });
});
