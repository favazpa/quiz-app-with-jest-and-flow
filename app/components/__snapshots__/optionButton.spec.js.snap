// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/options should render correctly 1`] = `
<TouchableOpacity
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#d8e5eb",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 60,
        "marginHorizontal": 20,
        "marginVertical": 5,
        "paddingHorizontal": 20,
      },
      {
        "backgroundColor": "orange",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "orange",
          "borderRadius": 25,
          "height": 45,
          "justifyContent": "center",
          "width": 45,
        },
        {
          "backgroundColor": "cyan",
          "borderColor": "yellow",
          "borderWidth": 1,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      NaN
    </Text>
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "700",
      }
    }
  />
</TouchableOpacity>
`;
